aggregator/src/aggregator.rs

```diff
@@ -3325,6 +3355,38 @@ impl VdafOps {
     }
 }
 
+fn write_task_aggregation_counter<C: Clock>(
+    datastore: Arc<Datastore<C>>,
+    shard_count: u64,
+    task_id: TaskId,
+    counters: TaskAggregationCounter,
+) {
+    // We write task aggregation counters back in a separate tokio task & datastore transaction,
+    // so that any slowness induced by writing the counters (e.g. due to transaction retry) does
+    // not slow the main processing. The lack of transactionality between writing the updated
+    // aggregation job & updating the counters means that process failure may cause us to leave
+    // some counter updates unaccounted for, but that is an acceptable tradeoff.
+    let ord = thread_rng().gen_range(0..shard_count);
+    tokio::task::spawn(async move {
+        let rslt = datastore
+            .run_tx("aggregate_continue_counters", |tx| {
```

Comment:
This transaction name is misleading, since it is used in all three aggregation code paths.
```suggestion
            .run_tx("aggregate_counters", |tx| {
```

Reply:
nit: I might even call it `update_task_aggregation_counters` to align with `update_task_upload_counters`.